packages/fuel-indexer/src/executor.rs

```diff
@@ -683,38 +728,32 @@ impl Executor for WasmIndexExecutor {
         let ptr = arg.get_ptr();
         let len = arg.get_len();
 
-        let res = timeout(
-            Duration::from_secs(self.timeout),
-            spawn_blocking({
-                let store = self.store.clone();
-                move || {
-                    let mut store_guard =
-                        tokio::runtime::Handle::current().block_on(store.lock());
-                    fun.call(&mut store_guard, ptr, len)
-                }
-            }),
-        )
-        .await;
-
-        match res {
-            Err(e) => {
-                error!("WasmIndexExecutor handle_events timed out: {e:?}.");
-                let _ = self.db.lock().await.revert_transaction().await?;
-                return Err(IndexerError::from(e));
+        let res = spawn_blocking({
+            let store = self.store.clone();
+            move || {
+                let mut store_guard =
+                    tokio::runtime::Handle::current().block_on(store.lock());
+                fun.call(&mut store_guard, ptr, len)
             }
-            Ok(Err(e)) => {
-                error!("WasmIndexExecutor handle_events failed: {e:?}.");
+        })
+        .await?;
+
+        if let Err(e) = res {
+            // get_instance_metering_points panics if metering is not enabled.
+            if self.metering_points.is_some()
```

Comment:
I think this can be changed a bit to be more readable. This `if` extends for 4 lines 😅